server/apiRoutes/post.js

```diff
@@ -0,0 +1,80 @@
+const router = require('express').Router();
+// const { client, cacheDuration } = require('../redis');
+const { Post } = require('../db/index');
+
+// create an event
+router.put('/createPost', async (req, res, next) => {
+    try {
+        res.json(await Post.create({
+            ...req.body,
+            userId: req.user.id,
+            groupId: req.params.groupId,
+        }));
+    } catch (err) {
+        next(err);
+    }
+});
+
+// grab a single post i.e. when someone clicks on a post link to read comments
+router.get('/:postId', async (req, res, next) => {
```

Comment:
This route needs to go below all other get/ routes. /:postId will catch anything and everything after the /.